src/sql/workbench/contrib/notebook/browser/notebookLinkHandler.ts

```diff
@@ -53,8 +53,17 @@ export class NotebookLinkHandler {
 			this.isMarkdown = this._link.attributes['is-markdown']?.nodeValue === 'true' ? true : false;
 			this.isEncoded = this._link.attributes['is-encoded']?.nodeValue === 'true' ? true : false;
 			this._isFile = this._link.protocol === `${Schemas.file}:` || this._link.protocol === `${Schemas.vscodeFileResource}:`;
-			// Given an anchor element for windows href link will need to use nodeValue instead as that does not encode the url
-			if (isWindows) {
+			// When editing on richtext, for local files paths use pathname to get the filepath of the link
+			// since the nodeValue results are ambiguous depending on OS ex: '/PathToParentFolder/ ./fiename.ipynb'.
+			if (this._isFile) {
+				// check if nodevalue is valid
+				if (notebookUtils.isFilepathValid(this._link.attributes['href']?.nodeValue)) {
```

Comment:
I'm a little confused by the usage of "valid" here. Because you could very well have a path that doesn't exist currently but is "valid". When wouldn't the file path be "valid" and why would you want to use the pathname directly in that case?